File: src/main/java/com/sothawo/springdataelastictest/movies/Movie.java
/*
 * (c) Copyright 2019 sothawo
 */
package com.sothawo.springdataelastictest.movies;

import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Document(indexName = "movies")
public class Movie {
	@Id private String id;
	private String title;
	@Field(type= FieldType.Integer)
	private Integer year;
	private List<String> cast;
	private List<String> genres;

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public Integer getYear() {
		return year;
	}

	public void setYear(Integer year) {
		this.year = year;
	}

	public List<String> getCast() {
		return cast;
	}

	public void setCast(List<String> cast) {
		this.cast = cast;
	}

    public List<String> getGenres() {
        return genres;
    }

    public void setGenres(List<String> genres) {
        this.genres = genres;
    }
}

File: src/main/kotlin/com/sothawo/springdataelastictest/Movie.kt
/*
 * (c) Copyright 2019 sothawo
 */
package com.sothawo.springdataelastictest

import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.Document
import org.springframework.data.elasticsearch.annotations.Field
import org.springframework.data.elasticsearch.annotations.FieldType

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
@Document(indexName = "movies")
class Movie {
    @Id
    var id: String? = null
    var title: String? = null
    @Field(type = FieldType.Integer)
    var year: Int? = null
    var cast: List<String>? = null
    var genres: List<String>? = null
}
